enhancements/network/ovs-dynamic-cpu-pinning.md

```diff
@@ -0,0 +1,199 @@
+
+Typically the users want to minimize the amount of reserved CPUs and leave 
+as many available CPUs as possible for business-related workloads.
+
+We have seen in some customer cases that it is possible to disrupt both the node
+and the cluster when the networking load rises, because OVS is CPU bound by the 
+limited number of system-reserved cores, and networking performance suffers.
+
+### User Stories
+
+> As an Openshift cluster administrator
+> I want to maximize my network dataplane bandwidth to all available cores in a dynamically reserved CPU environment
+> so that the cluster can endure a high network workload.
+
+### Goals
+
+Every cluster node that is configured with static CPU manager policy and
+a low number of `reservedSystemCpus` does not experience any excessive latency 
+increase while dealing with high network loads.
+
+### Non-Goals
+
+The logic used by Node Tuning Operator to decide if and when to enable the feature
+are out of this enhancement's scope.
+
+## Proposal
+
+As of today, a set of CPUs can be reserved for system workload (i.e. other than Kubernetes workload)
+by specifying the `--reserved-cpus` kubelet parameter (or `reservedSystemCpus` via configuration file).
+
+The main idea to resolve the issue described above is to allow OVS 
+to spread out to other cpus outside the reserved cpu pool. However 
+a care must be taken to not interfere with cpus that are assigned 
+to latency sensitive workloads.
+
+### Workflow Description
+
+The feature is triggered by [Node Tuning Operator](https://github.com/openshift/cluster-node-tuning-operator) (NTO) when it detects the 
+performance conditions to turn it on. The logic NTO uses to activate this 
+feature is out of this enhancement's scope. 
+The following steps will be implemented to turn the feature on:
+- NTO's daemonset creates a file in `/etc/openvswitch/enable_dynamic_cpu_affinity` 
+  on every node where it wants to enable the pinning feature.
+- The file is used by ovnkube-node pod to start a goroutine that constantly monitors 
+  the affinity of itself and OVS daemons, aligning the latter if 
+  needed (see [CPU affinity alignment](#cpu-affinity-alignment)). The file is the signal for ovnk to enable 
+  the feature (see [Activation file](#activation-file))
+
+```mermaid
+sequenceDiagram
+    Note right of NodeTuningOperator: via MachineConfig
+  
+    NodeTuningOperator->>ovs-vswitchd: set_slice
+    NodeTuningOperator->>ovsdb-server: set_slice
+    NodeTuningOperator->>/etc/openvswitch/enable_dynamic_cpu_affinity: fwrite('1')
+    activate /etc/openvswitch/enable_dynamic_cpu_affinity
+    ovnkube-node ->> /etc/openvswitch/enable_dynamic_cpu_affinity: fstat()
+    
+    activate ovnkube-node
+    loop
+    ovnkube-node ->> ovnkube-node: x = sched_getaffinity(x)
+    ovnkube-node ->> ovs-vswitchd: sched_setaffinity(x)
```

Comment:
sched_setaffinity(OVS) - the important note here is that it might take few seconds for this to happen. First because cpu manager reconciler will take few seconds to update the affinity of the ovnkube-node container and then another few seconds until ovn-kube notices and updates the affinity of OVS. During this window a newly started guaranteed and cpu pinned container might encounter latency spikes.

Reply:
Thanks for these details. Adding to the `CPU affinity alignment` section